sass/tokens/_parse.scss

```diff
@@ -232,28 +251,43 @@ $_stack: ();
 
   // replace alias values in a string
   @else if ($type == 'string') {
-    $alias: has-alias($value);
+    $alias: next-alias($value);
 
     @while $alias {
-      $lookup: lookup-alias($map, $alias);
+      $alias-key: map.get($alias, 'alias');
+      $lookup: lookup-alias($map, $alias-key);
 
-      // fail on missing keys
+      // If lookup fails, it could be a missing full alias, or a color, or a missing
+      // sub-alias. We don't distinguish between the last 2, but return null for missing
+      // full alias (eg value is '#foo->not-here').
       @if (meta.type-of($lookup) != 'map') {
-        @return $lookup;
+        @if (1 == map.get($alias, 'start')) and
+          (string.length($alias-key) + 2 == map.get($alias, 'end'))
+        {
```

Comment:
Could this be a comparison of `$value == '##{$alias-key}'`? That seems a bit more clear and to the point?

Reply:
Yes, I agree. Will update. Thanks!